src/middleware/error_handling.ts

```diff
@@ -70,11 +79,15 @@ export function errorHandler(
     if (res.headersSent) {
         return next(err);
     }
-    if ((err as any).isAPIError || (err as any).statusCode) {
```

Comment:
Moved all the `if` statements into `generateError` so we can reason about error cases in one function. It SHOULD be the same logic with the added fallback case of `500`.